### ISSUE TITLE
Question on privacy

### ISSUE THREAD

**Issue by IzzySoft** [deltazefiro/Amarok-Hider#2]
As your app's description speaks of privacy: would you mind removing/replacing the MS Analytics trackers? MS is not exactly known for being a privacy defender, and the presence of those trackers might scare away some users. If you think you urgently need some analytics, you can find [some alternatives here](https://gitlab.com/-/snippets/1896503#acceptable-analytics).

Thanks for considering!

**Comment by deltazefiro** [comment]
Thanks for the reminder. I will remove the Analytics tracker.  

But Amarok still depends on Appcenter Distribution to update and Crashes to diagnose unexpected errors. The alternatives you provided are not available in China due to the [GFW](https://en.wikipedia.org/wiki/Great_Firewall).

**Comment by IzzySoft** [comment]
> depends on Appcenter Distribution to update

Not really needed, is it? For updates there are the stores/repositories. Your app can e.g. be installed [from my repo](https://apt.izzysoft.de/fdroid/index/apk/deltazero.amarok) – which is usually done using one of the available [F-Droid clients](https://android.izzysoft.de/applists/category/named/apps_markets#group_1181) which then also takes care for updating.

> and Crashes to diagnose unexpected errors

Alternatives to that part are available, as pointed out – and the list is not limited to those. Hm, GFW might be an issue then I had not thought of indeed. A possible solution could be (if you'd go that far) to have an additional build flavor: `foss` without the MS libs and using a FOSS analytics library, and `gfw` for the "blocked area". If you'd attach both APKs to your releases, I'd configure my updater to fetch the `foss` one, while folks in China could pick the `gfw` one?

And thanks for taking care!

**Comment by deltazefiro** [comment]
> For updates there are the stores/repositories. Your app can e.g. be installed from my repo

Your F-droid repo is awesome and I am delighted that users can install Amarok from it! But, to be honest, there is really few people in China using F-Droid. Amarok is still at the early beta stage so currently I have no plans to upload it to stores like Google Play. Thus, I think using MS appcenter's in-app update is the most convenient way.

> A possible solution could be (if you'd go that far) to have an additional build flavor

I am a newbie in Android developing and adding the lib switch is kinda out of my reach. But instead, I think I can remove the `appcenter app secret`, which is the app identifier, from the source code and inject it with CI while compiling. F-droid server would compile the APK itself, right? It would not inject the secret and without the identifier, MS appcenter lib would not work.

**Comment by kekkodance** [comment]
Maybe updates could be fetched from GitHub releases (not blocked in China)?

**Comment by IzzySoft** [comment]
I guess. That's btw where my repo updater fetches them from. Using an F-Droid client of choice (see above), that could be used for not only this app (I don't think my servers are blocked by the GFW, they're not "big enough" to warrant that), almost 1k apps in there.

**Comment by deltazefiro** [comment]
@kekkodance Actually, Github is "half blocked" in China (with DNS cache pollution), which means viewing text on Github is OK but downloading files from the release is unbearably slow in some areas.

**Comment by deltazefiro** [comment]
@IzzySoft 
> of course if time permits

That's the point. I am terribly busy nowadays. But I suppose I can have a try on version 0.8.1b (the next release 0.8.0b is the first version of 0.8 and features several unstable modules, I would rather keep the diagnosis).  

One more question: 

> my repo instead takes the APKs you release.

How can you fetch the `foss` version after I set up the build flavors? Since your server does not build, should I put the `foss` version apk in release assets?

**Comment by IzzySoft** [comment]
Yes, that naming would help a lot! I could then tell my updater to only pull APKs matching that naming pattern (here: `/release-foss/i`). If the package name (aka `applicationId`) and signing remains the same, it would be seamless updates for the users.

**Comment by deltazefiro** [comment]
Apologies for the late response, but I have finally done it. 
In the lastest pre-release ([v0.8.1b1](https://github.com/deltazefiro/Amarok-Hider/releases/tag/v0.8.1b1)), there is an apk with `-foss` suffix.  
I've removed all MS libs in that version.  

>  If the package name (aka applicationId) and signing remains the same, it would be seamless updates for the users.

The `foss` version have to use a different PackageName – to prevent confusion – so unfortunately, they need to uninstall the pervious version first to install the new one.

**Comment by IzzySoft** [comment]
@deltazefiro Thanks a lot! I can confirm the "offenders" being gone with the `foss` variant. I accept and understand your decision for a separate applicationId, though I think it not strictly necessary. I hence moved `deltazero.amarok` to `deltazero.amarok.foss` in my repo, which will be rolled out with the next sync around 7 pm UTC. So if you want, be welcome to [pick a badge/shield](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to [link there](https://apt.izzysoft.de/fdroid/index/apk/deltazero.amarok.foss) as soon as it showed up :smiley:

**Comment by IzzySoft** [comment]
@deltazefiro as your app is listed at F-Droid.org now (congrats!), do you still want me to keep it in my repo – or shall I remove it (after a decent overlap; I usually remove apps after 10..14 days unless explicitly requested to keep)? I see the badge linking to my repo is still there in the Readme, so I thought I ask (and btw, F-Droid has the same badge API available as well, if you want to add another badge).

**Comment by deltazefiro** [comment]
Hey @IzzySoft!
I think it's alright to keep it - thanks to the reproducible build  :P 
But would you mind syncing the current Fastlane to your repo? The original one is out-of-date. Thanks!

**Comment by IzzySoft** [comment]
> I think it's alright to keep it there - thanks to the reproducible build :P

Full ack! So folks can install from F-Droid (for safety/trust) and enjoy the faster updates via my repo :smiley:

> But would you mind syncing the current Fastlane to your repo?

:open_mouth: No Fastlane configured at my end? Oh my… Fixing that right now. Question on the screenshots: what value do those frames add? I just see a disadvantage there (on small screens they waste valuable screen estate, thus making it harder to see the details). Mind using "unframed" ones? I've now enabled everything but screenshots here, waiting for your reply. Oh, and of course I marked your app to stay. On the website of my repo this will be reflected on the details page with a small banner linking here:

![image](https://user-images.githubusercontent.com/6781438/216768435-c7349d71-cc5d-47e1-bd9c-db0eeaa06a3a.png)

Changes will be reflected with the next sync, as usual, around 7 pm UTC.

**Comment by deltazefiro** [comment]
> Full ack! So folks can install from F-Droid (for safety/trust) and enjoy the faster updates via my repo smiley

Wow that's a good idea! And better, would it be possible to set your updater to fetch the pre-release as well?

> Question on the screenshots: what value do those frames add? I just see a disadvantage there (on small screens they waste valuable screen estate, thus making it harder to see the details). Mind using "unframed" ones?

Thanks for the concern. But I'd like to use the framed ones. They give me a kinda feeling of, uh, looking at  "full screenshots" rather than "parts". :smile: Anyway, thanks!

**Comment by IzzySoft** [comment]
> And better, would it be possible to set your updater to fetch the pre-release as well?

Done.

> They give me a kinda feeling

Well, consider me a patron of those not able to afford the always latest and greatest device – plus those using their devices as long as possible¹ (sustainability), so I had to ask. Including the notification bar and the navigation bar as well (while not framing) should be sufficient to get the feeling there's nothing amiss, IMHO. But I of course respect your decision – so screenshots are enabled now as well. Up to you now to give the "low-end fraction" more weight.

> Anyway, thanks!

Anytime!

----

¹ <sub>My oldest device still-in-use is a 2010 Motorola Milestone 2 (CM with Kitkat), which serves as night-stand only. The next "younger ones" are a 2015 Fairphone 2 (personal device, Android 11 with LOS) and a 2016 Wileyfox Swift (Android 10/LOS, used for reviews/on-device-testing of F-Droid applicants). The newest is a 2020 SHIFT6mq (Android 10/ShiftOS-L; that's their official (!!) ROM without GApps) being my main driver and "business phone".</sub>

**Comment by deltazefiro** [comment]
> Well, consider me a patron of those not able to afford the always latest and greatest device – plus those using their devices as long as possible¹ (sustainability), so I had to ask.

Ah, apologies that I didn't consider that. Personally, I did not own a mobile device until three years ago – and I think I missed the era of small screens. :disappointed:
However, I would like to insist on using the original framed images in the F-droid main repo. But I also created an [unframed version](https://mega.nz/file/iXQzxYwZ#lHyUUkYyHkfE8pR2YvJPBJCib9FzINy3vZGQ2RALQFg) for your repository! :laughing:

**Comment by IzzySoft** [comment]
Your frames are not too much waste (I would insist on unframed where the frame makes up 1/3 of the entire image, here it's quite decent so I just suggested). I prefer to have things updated automatically via Fastlane instead of getting a separate set per link to some share-hoster :wink:

And yes, I'm quite proud of my device collection. The Milestone needed a replacement battery last year as the original one did swell (after 12 years! That was quality!), so did the FP2 (after 7 years) and so will the Fox soon (also after 7 years), but all of them run fine. None of them ever was some crazy "flagship device", all of them middle-class. But it seems I made the right choices for sustainable devices. Only exception so far: my BQ Aquaris X5 Plus. Battery swollen, but not replaceable. Tried replacing it nevertheless, couldn't get the new battery working (probably did not connect correctly, or I got a faulty battery replacement, don't know). That was the first and the last phone I bought without a replaceable battery.

**Comment by deltazefiro** [comment]
> Your frames are not too much waste (I would insist on unframed where the frame makes up 1/3 of the entire image, here it's quite decent so I just suggested). I prefer to have things updated automatically via Fastlane instead of getting a separate set per link to some share-hoster wink

Well then. Let's close this issue :wink: 

> And yes, I'm quite proud of my device collection. The Milestone needed a replacement battery last year as the original one did swell (after 12 years! That was quality!), so did the FP2 (after 7 years) and so will the Fox soon (also after 7 years), but all of them run fine. None of them ever was some crazy "flagship device", all of them middle-class. But it seems I made the right choices for sustainable devices. Only exception so far: my BQ Aquaris X5 Plus. Battery swollen, but not replaceable. Tried replacing it nevertheless, couldn't get the new battery working (probably did not connect correctly, or I got a faulty battery replacement, don't know). That was the first and the last phone I bought without a replaceable battery.

Awesome! I'm quite interested in playing with old devices - though I do not have one. I enjoy browsing through the posts about modifying old devices on Coolapk (a Chinese smartphone community, just like xda). They replaced everything to make devices sustainable: from batteries to screens :laughing:

**Comment by IzzySoft** [comment]
> They replaced everything to make devices sustainable: from batteries to screens :laughing:

Owning a Fairphone 2 and a SHIFT6mq, I find that normal. Actually, my good old Fairphone 2 consists of part from multiple sources: I got it for free from a friend who switched to the FP4, but the camera was broken – so I got a 8 MPx camera module from another friend who had replaced his by a 24 MPx. Then I found out the microphone didn't work, so a third gave me a spare part ("bottom module") to get that fixed. No repair shop needed, there are even official guides out there on how to do it; each replacement was done within a few minutes. SHIFT even offers courses if you want to become a "certified repair person", both have a great community and SHIFT even a phantastic pair of CEOs (called the "SHIFT brothers" – and yes, they are indeed brothers, presenting news and going-ons in their "brother talk" videos – which are always heart-warming).

But yes, not a matter of principle everywhere, unfortunately. And I dislike it very much that when the battery blew you cannot use the device without one in most cases (e.g. as older test device, stationary camera, media server, whatever). Just yesterday I found an article describing how to trick them into that state and "fake" a battery. Might play with that :wink:
